lib/agent.js

```diff
@@ -44,12 +46,6 @@ function Agent () {
   this.lambda = lambda(this)
 }
 
-Object.defineProperty(Agent.prototype, 'logger', {
```

Comment:
I've updated the only two places `agent.logger` needs to be set. This avoids runtime code needing to go through an getter function for every `agent.logger...` call. Theoretically that should be an ever so slight perf win. :)  
(This originated from an earlier stab at this that removed even having `.logger` on the "Config" object -- since backed out.)